api/repositories/route_repository.go

```diff
@@ -421,6 +455,19 @@ func (r *RouteRepo) PatchRouteMetadata(ctx context.Context, authInfo authorizati
 	return cfRouteToRouteRecord(*route), nil
 }
 
+func (r *RouteRepo) GetState(ctx context.Context, authInfo authorization.Info, spaceGUID string) (ResourceState, error) {
```

Comment:
From my point of view, the logic to check whether the space contains unmapped routes should not be here. Whether a space contains unmapped routes or not is orthogonal to whether the space itself is ready or not.

How about we have a special handling of the `SpaceDeleteUnmappedRoutesType` job type in the job handlers, similar to the [special handling of applying space manifest](https://github.com/cloudfoundry/korifi/blob/b170c35905716ab3bb577122950180f132ad8b84/api/handlers/job.go#L87). I.e.

```
	switch job.Type {
	case syncSpaceJobType:
		// ....
	case SpaceDeleteUnmappedRoutesType:
		// list unmapped routes in that space via `routeRepo.ListRoutes(umapped=true)`. If the slice is empty, return "ready", "unknown" otherwise
	default:
		// ....
	}
```